src/Analyzer/Passes/QueryAnalysisPass.cpp

```diff
@@ -3769,8 +3776,23 @@ IdentifierResolveResult QueryAnalyzer::tryResolveIdentifier(const IdentifierLook
 
     if (!resolve_result.resolved_identifier && identifier_lookup.isTableExpressionLookup())
     {
-        auto cte_query_node_it = scope.cte_name_to_query_node.find(identifier_lookup.identifier.getFullName());
-        if (cte_query_node_it != scope.cte_name_to_query_node.end())
+        auto full_name = identifier_lookup.identifier.getFullName();
+        auto cte_query_node_it = scope.cte_name_to_query_node.find(full_name);
+
+        /// CTE may reference table expressions with the same name, e.g.:
+        ///
+        /// WITH test1 AS (SELECT * FROM test1) SELECT * FROM test1;
+        ///
+        /// Since we don't support recursive CTEs, `test1` identifier inside of CTE
```

Comment:
If/When we support it, will we need to change the behavior? Or will it be easy because recursive CTEs have a special syntax with a keyword, and the behavior of existing queries won't change?

Reply:
I suppose, yes, but I don't think this patch would introduce any complications in the future.